apps/api-documenter/src/markdown/MarkdownEmitter.ts

```diff
@@ -115,7 +115,7 @@ export class MarkdownEmitter {
         if (context.insideTable) {
           const code: string = this.getTableEscapedText(docCodeSpan.code);
           const parts: string[] = code.split(/\r?\n/g);
-          writer.write(parts.join('`<p/>`'));
+          writer.write(parts.join('</code><br/><code>'));
```

Comment:
I agree that you've found the best possible solution, but I'll point out that it has the downside that each line of code will have a separate styling rectangle. It looks okay on GitHub, but results may vary with other Markdown renders.

I've never liked the pipes-and-dashes table construct.  It has many limitations, and the syntax isn't particularly easy to write.  If I remember right, [CommonMark](https://commonmark.org/) chose not to standardize the pipes-and-dashes table after a very long discussion about it.  My general philosophy is that while Markdown saves a lot of typing for basic formatting problems, we should defer to using HTML for any complex nesting structures.  (This is the philosophy that I'm advocating for TSDoc.)  

Unfortunately, as soon as we introduce an HTML table tag, it turns off Markdown entirely:

```markdown
<table><tr><td>
`code`
**bold**
</td></tr></table>
```

gets rendered as:

<table><tr><td>
`code`
**bold**
</td></tr></table>

Long term, I'd like to add an `api-documenter html` command line that generates (style-neutral) HTML instead of Markdown.  HTML is much easier to generate, whereas Markdown was intended for an interactive human editing experience.  Markdown is very difficult for a machine to generate correctly as we see in this PR.  The main reason we chose Markdown initially was to support a custom documentation engine that doesn't accept HTML input.  This is a fairly common situation, though, so **api-documenter** should certainly continue to support Markdown forever, alongside any other output types we may add.

Another idea I've considered is finding an existing documentation generator with very professional templates, and writing an adapter to drive it using .api.json files.  I wouldn't want to bake too much fancy rendering into **api-documenter**, because its goal is to be complete but simple and readable, so people can use its source code as a guide when writing their own custom solution.

Reply:
I'm glad to hear that Markdown will be supported going forward. There are still good reasons to support it fully:

- markdown allows documentation to be viewed in GitHub repositories naturally. You can link to documentation from a README and it will render nicely.
- markdown is common as an input format for static site generators commonly used for documentation. For example, docusaurus can consume api-documenter output and render html with the same style as other pages it generates. This makes for a much nicer documentation site.

If tables turn out to be a continuing source of trouble, maybe there's a way to generate reasonable output without tables. In any case, while I agree that html is better specified and would make it easier to generate, I'd vote for keeping markdown for its interoperability.

Reply:
Thanks @acchou, useful feedback!

Reply:
Btw do you have an example of docusaurus output? Maybe we should mention this option in the docs.

Reply:
I'm working on my site right now, I will post a link here when it's published.

Reply:
Take a look at my site (the actual project is not yet published on npm, will happen in the next few days at latest): https://faastjs.org. The api-documenter generated part is under the [API link](https://faastjs.org/docs/api/faastjs).

I did need to write a script to make docusaurus style the markdown files. Basically the script does the following for `api/*.md` generated by api-documenter:
- Removes the "Home" breadcrumb link, which isn't configurable and isn't needed since docusaurus renders the markdown file with a navigation bar at the top and a footer at the bottom.
- Adds a header to each file with the following format:
```
---
id: <ID>
title: <TITLE>
hide_title: true
---
```

The `hide_title` part instructs docusaurus to not insert the title itself, since it's already in the markdown.

Reply:
Wow @acchou , this looks pretty good!  I wonder if we could make any improvements to **api-documenter** for better compatibility with docusaurus.

Reply:
Well, if api-documenter allowed some configuration and/or omission of the home link in the breadcrumb trail, that would help. It's an awkward link because it assumes you have an index.html file in a specific location.

The header is a bit more docusaurus-specific... maybe if there were some way to add a plugin for various site generators, it could be made easier. But the script wasn't hard as it is. It's just not out-of-the box easy. From my perspective I would rather the limited resources go into other things, for example:

- watch mode so docs can be regenerated incrementally
- links for types. Currently the workaround is to scatter @link tags everywhere, but it would be much nicer if, for example, [faast()](http://localhost:3000/docs/api/faastjs.faast) had links for `Provider` and `CommonOptions` in the type field, instead of my having to manually add it to the description.
- more work on the documentation site so it would be more obvious to people that api-documenter exists. It took effort to realize it could do exactly what I wanted!

Pretty sure these are already on your radar...

Reply:
The type links are definitely a priority. Someone already started a PR (although it hasn't been worked on in a while).

Regarding the docs, we just [updated the web site](https://api-extractor.com/pages/setup/generating_docs/) when AE7 was released. If there are topics you want to see that still aren't covered by the current docs, let me know.

Reply:
(I'm also curious whether you will try to implement a navigation tree for your docs. Markdown by itself doesn't provide any facility for this, but we could easily write the tree into a JSON file similar to what we do for DocFX. In that case you might ask to disable the breadcrumb entirely.)

Reply:
A navigation tree might be nice, but I don't consider it a priority until it's clear the existing breadcrumbs are insufficient. Plus I plan on adding search to the site so that should alleviate some of the need.

Another way to mitigate the demand for navigation would be to reorganize the output so there aren't so many files generated. It would be nice if every class or module was a single file. There are many pages that just have a short description and little else.

Reply:
BTW I added your link as an example for the [Generating API docs](https://api-extractor.com/pages/setup/generating_docs/) article

Reply:
Cool, thanks!

Also, to drive home the "Home" link issue, note that that node-core-library documentation's breadcrumb for "Home" is a 404.

Reply:
@acchou thanks for reporting this!  

The missing extension works okay [on GitHub pages](https://microsoft.github.io/web-build-tools/api/node-core-library.html) but I didn't realize that it was broken in the GitHub markdown viewer.  Here's a PR to fix it: https://github.com/Microsoft/web-build-tools/pull/1251